src/middlewared/middlewared/main.py

```diff
@@ -144,16 +144,18 @@ def _load_apis(self) -> dict[str, API]:
     def _load_api_versions(self) -> [APIVersion]:
         versions = []
         api_dir = os.path.join(os.path.dirname(__file__), 'api')
-        for version_dir in sorted(pathlib.Path(api_dir).iterdir()):
-            if version_dir.name.startswith('v') and version_dir.is_dir():
-                version = version_dir.name.replace('_', '.')
-                self._console_write(f'loading API version {version}')
-                versions.append(
-                    APIVersion.from_module(
-                        version,
-                        importlib.import_module(f'middlewared.api.{version_dir.name}'),
-                    ),
-                )
+        api_versions = [
+            (version_dir.name.replace('_', '.'), f'middlewared.api.{version_dir.name}')
+            for version_dir in sorted(pathlib.Path(api_dir).iterdir())
```

Comment:
Let's use `os.scandir` here since we're trying to squeeze as much performance out as possible.